### DISCUSSION TITLE
Подсовка DNS

### DISCUSSION THREAD

**Discussion by ghost** [hufrea/byedpi#63] [Ideas]
Есть ли шанс, что будет добавлена возможность задавать программе dns, к которому она будет направлять указанные в списке сайты, как в goodbyedpi? Ваша программа даже на андроиде спокойно работает, но указанной функции очень не хватает

**Comment by hufrea** [comment]
Вряд ли, DNS сервер можно поменять средствами системы или в самом ПО, Firefox и Chrome позволяют использовать DoH, Android с 9-й версии тоже поддерживает шифрованный DNS.

**Comment by ghost** [comment]
Пихать днс на всю систему это крайне сомнительная затея, как по мне, и намного удобнее было бы раздавать его на конкретные адреса через программу, да и в тот же клиент ютуба его не впихнёшь

**Comment by hufrea** [comment]
> Пихать днс на всю систему это крайне сомнительная затея

А мне кажется хорошая, учитывая, что обычный DNS over UDP не шифруется.

> в тот же клиент ютуба его не впихнёшь.

Клиент будет использовать системный DNS, или DNS VPN-а, так что это не должно быть проблемой.

**Comment by ghost** [comment]
Прописанный в систему днс будет влиять на все соединения, а не только те, что нужны, и это может плодить проблем. В прокси же оно будет только на то куда указано цепляться, тем более если в какой-то из программ доступен ручной ввод прокси, либо если тот же клиент ютуба через хитрость можно будет обернуть под прокси. Да и тем более что программа-то не только на андроиде используется. На том же гудбайдипиай днс редирект это прям мастхэв, иначе бы не писал сюда с этим вопросом

**Comment by rootk1t** [comment]
> Прописанный в систему днс будет влиять на все соединения, а не только те, что нужны, и это может плодить проблем. В прокси же оно будет только на то куда указано цепляться, тем более если в какой-то из программ доступен ручной ввод прокси, либо если тот же клиент ютуба через хитрость можно будет обернуть под прокси. Да и тем более что программа-то не только на андроиде используется. На том же гудбайдипиай днс редирект это прям мастхэв, иначе бы не писал сюда с этим вопросом

Думаю, что автор byedpi прав.
В домашней сети можно единожды прописать шифрованные DNS (DOT/DOH) на роутере, чтобы все подключенные устройства их использовали. 
Вне домашней сети, прописать шифрованные DNS (DOT/DOH) в настройках сетевого подключения или системы в целом, если речь про десктопные OS.
На андроиде в обоих случаях не помешает прописать шифрованный DNS в настройках системы, чтобы провайдер не подменял их на свои.

Гудбайдипиай рассчитан на другую аудиторию, которой требуется решение в 1 клик, и которая не знает или не хочет знать о настройках DNS/прочего, а просто хочет нажать 1 кнопку и чтобы всё работало.
